components/brave_wallet_ui/page/screens/composer_ui/from_asset/from_asset.tsx

```diff
@@ -124,8 +158,14 @@ export const FromAsset = (props: Props) => {
   const tokenBalance =
     !account || !token || !tokenBalancesRegistry
       ? ''
+      : token.coin === BraveWallet.CoinType.BTC && bitcoinBalances
+      ? bitcoinBalances.availableBalance
       : getBalance(account.accountId, token, tokenBalancesRegistry)
 
+  const hasPendingBalance = new Amount(
+    bitcoinBalances?.pendingBalance ?? ''
+  ).gt(0)
+
```

Comment:
That's ok for now as it needs to get into beta, but in general it is not scalable. Assume that everywhere we are adding special handling of BTC balance we will also need some similar ZCash additions.

Instead the only entrypoint for balance should be getBalance call which should return composite balance(`total`, `pending`, `available` etc). All coins we had(ETH, SOL, FIL) just fill `total` and `available`(=`total`). New coins also may fill everything else. All components presenting balance to user should be able to handle all needed fields.

Reply:
The `getBalance` here is not a call, is a `utility` helper that takes in the `tokenBalancesRegistry` and returns the balance for an `accountId`. Currently we are settings `totalBalance` for `Bitcoin` in the registry https://github.com/brave/brave-core/blob/master/components/brave_wallet_ui/common/slices/endpoints/token_balances.endpoints.ts#L543 which is global for the wallet, is not specific for the `Send` screen. I agree though, lets brainstorm on a reusable `helper` to use once `ZCash` is ready!